File: src/sys/Aluno.java
package sys;

import java.util.LinkedList;

public class Aluno {
	
	public static int NUM_NOTAS = 3;
	public static int MATRICULA_COUNT = 1;
	
	private static String geradorDeMatricula() {
		String matricula = String.format("%08d", Aluno.MATRICULA_COUNT);
		Aluno.MATRICULA_COUNT++;
		return matricula;
		
	}
	
	private String matricula;
	private String nome;
	private String nascimento; //data de nascimento: DD/MM/AAAA
	private LinkedList<Materia> materias;
	
	public Aluno(String nome, String nascimento) {
		this.nome = nome;
		this.nascimento = nascimento;
		this.matricula = Aluno.geradorDeMatricula();
		this.materias = new LinkedList<Materia>();
		for(NomesMateria m: NomesMateria.values()) {
			this.materias.add(new Materia(Aluno.NUM_NOTAS, m));
		}
	}
	
	public Aluno( String matricula, String nome, String nascimento) {
		this.nome = nome;
		this.nascimento = nascimento;
		this.matricula = matricula;
		for(NomesMateria m: NomesMateria.values()) {
			this.materias.add(new Materia(Aluno.NUM_NOTAS, m));
		}
	}
	
	public void cadastrarNota(NomesMateria materia, double nota, int unidade) {
		for(Materia m: this.materias) {
			if(m.getNome().equals(materia)) {
				m.cadastrarNota(nota, unidade);
			}
		}
	}
	
	public LinkedList<Materia> exibirNotas(){
		return this.materias;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getNascimento() {
		return nascimento;
	}

	public void setNascimento(String nascimento) {
		this.nascimento = nascimento;
	}

	public String getMatricula() {
		return matricula;
	}
	
	public String toString() {
		return "Aluno: "+this.nome+", Matrícula: "+this.matricula;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((matricula == null) ? 0 : matricula.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Aluno other = (Aluno) obj;
		if (matricula == null) {
			if (other.matricula != null)
				return false;
		} else if (!matricula.equals(other.matricula))
			return false;
		return true;
	}

}

File: srcPython/sys/aluno.py
import yaml
from SYS.Materias import NomesMateria,Materia

MATRICULA_COUNT = 0
NUMERO_DE_NOTAS = 3

def geradorDeMatricula():
	global MATRICULA_COUNT
	MATRICULA_COUNT += 1
	return f'{MATRICULA_COUNT}'.zfill(4)

class Aluno(object):
	
	global NUMERO_DE_NOTAS

	def setMatriculaCount(quant:int):
		global MATRICULA_COUNT
		MATRICULA_COUNT = quant

	def getMatriculaCount():
		global MATRICULA_COUNT
		return MATRICULA_COUNT
	
	def __init__(self,nome:str, dataDeNascimento:str, materiasDumped = []):
		self.nome = nome
		self.dataDeNascimento = dataDeNascimento
		self.matricula = geradorDeMatricula()
		self.materias = list()
		if len(materiasDumped) == 0:
			for i in NomesMateria:
				self.materias.append(Materia(i,NUMERO_DE_NOTAS))
		else:
			for i in materiasDumped:
				materia = Materia(NomesMateria(i["nome"]),NUMERO_DE_NOTAS)
				materia.notas = i["notas"]
				self.materias.append(materia)

	def cadastrarNota(self, materia:NomesMateria, unidade:int, nota:float):
		for i in self.materias:
			if i.nome == materia:
				i.cadastrarNota(unidade, nota)
				break

	def __str__(self):
		return f"Aluno: {self.nome}, matricula: {self.matricula}"
	
	def __eq__(self,obj):
		if obj != None:
			if obj.__class__ == self.__class__:
				if obj.matricula == self.matricula:
					return True
		return False